Core/include/Acts/Utilities/Intersection.hpp

```diff
@@ -39,144 +39,210 @@ inline std::ostream& operator<<(std::ostream& os, IntersectionStatus status) {
 ///
 ///  Intersection struct used for position
 template <unsigned int DIM>
-struct Intersection {
+class Intersection {
+ public:
+  /// Position type
+  using Position = ActsVector<DIM>;
   /// Status enum
   using Status = IntersectionStatus;
 
-  /// Position of the intersection
-  ActsVector<DIM> position = ActsVector<DIM>::Zero();
-  /// Signed path length to the intersection (if valid)
-  typename ActsVector<DIM>::Scalar pathLength{
-      std::numeric_limits<double>::infinity()};
-  /// The Status of the intersection
-  Status status{Status::unreachable};
+  /// Default constructor
+  Intersection() = default;
 
   /// Constructor with arguments
   ///
-  /// @param sinter is the position of the intersection
-  /// @param slength is the path length to the intersection
-  /// @param sstatus is an enum indicating the status of the intersection
-  Intersection(const ActsVector<DIM>& sinter, double slength, Status sstatus)
-      : position(sinter), pathLength(slength), status(sstatus) {}
+  /// @param position is the position of the intersection
+  /// @param pathLength is the path length to the intersection
+  /// @param status is an enum indicating the status of the intersection
+  Intersection(const Position& position, double pathLength, Status status)
+      : m_position(position), m_pathLength(pathLength), m_status(status) {}
 
-  /// Default constructor
-  Intersection() = default;
+  /// Returns whether the intersection was successful or not
+  explicit operator bool() const { return m_status != Status::missed; }
 
-  /// Bool() operator for validity checking
-  explicit operator bool() const { return (status != Status::missed); }
+  const Position& position() const { return m_position; }
 
-  /// Smaller operator for sorting,
-  /// - it respects the validity of the intersection
-  /// @param si is the intersection for testing
-  bool operator<(const Intersection<DIM>& si) const {
-    if (status == Status::unreachable) {
-      return false;
-    }
-    // Now check the pathLength
-    if (si.status != Status::unreachable) {
-      return (pathLength < si.pathLength);
-    }
-    // The current one wins, no re-ordering
-    return true;
+  ActsScalar pathLength() const { return m_pathLength; }
+
+  Status status() const { return m_status; }
+
+  static bool forwardOrder(const Intersection& aIntersection,
+                           const Intersection& bIntersection) {
+    auto a = aIntersection.pathLength();
+    auto b = bIntersection.pathLength();
+    return a < b;
   }
 
-  /// Greater operator for sorting,
-  /// - it respects the validity of the intersection
-  /// @param si is the intersection for testing
-  bool operator>(const Intersection<DIM>& si) const {
-    if (status == Status::unreachable) {
+  static bool closestOrder(const Intersection& aIntersection,
```

Comment:
This method (and `forwardOrder`) could use a doc string describing the behavior. The name helps already but why not make it explicit?